src/Tests/dotnet-workload-install.Tests/WorkloadGarbageCollectionTests.cs

```diff
@@ -0,0 +1,393 @@
+
+            installer.GarbageCollect(getResolver);
+
+            foreach (var pack in packsToCollect)
+            {
+                PackShouldExist(pack, false);
+                PackRecord(pack, "6.0.300").Should().NotExist();
+            }
+            foreach (var pack in packsToKeep)
+            {
+                PackShouldExist(pack, true);
+                PackRecord(pack, "6.0.300").Should().Exist();
+            }
+
+            ManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300").Should().NotExist();
+            ManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300").Should().Exist();
+            ManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300").Should().NotExist();
+
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.100", "testmanifest", "1.0.0", "WorkloadManifest.json")).Should().NotExist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "2.0.0", "WorkloadManifest.json")).Should().Exist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "3.0.0", "WorkloadManifest.json")).Should().NotExist();
+        }
+
+        //  Additional scenarios to add tests for once workload sets are added:
+        //  Garbage collect workload sets
+        //  Garbage collect with install state with workload set
+        //  Don't garbage collect baseline workload set
+
+        void PackShouldExist(PackInfo pack, bool shouldExist)
+        {
+            if (pack.Kind == WorkloadPackKind.Library)
+            {
+                if (shouldExist)
+                {
+                    new FileInfo(pack.Path).Should().Exist();
+                }
+                else
+                {
+                    new FileInfo(pack.Path).Should().NotExist();
+                }
+            }
+            else
+            {
+                if (shouldExist)
+                {
+                    new DirectoryInfo(pack.Path).Should().Exist();
+                }
+                else
+                {
+                    new DirectoryInfo(pack.Path).Should().NotExist();
+                }
+            }
+        }
+
+        PackInfo CreatePackInfo(string id, string version, WorkloadPackKind kind, string resolvedPackageId = null)
+        {
+            if (resolvedPackageId == null)
+            {
+                resolvedPackageId = id;
+            }
+
+            string path;
+            if (kind == WorkloadPackKind.Library)
+            {
+                path = Path.Combine(_dotnetRoot, "library-packs", $"resolvedPackageId.{version}.nupkg");
+            }
+            else
+            {
+                path = Path.Combine(_dotnetRoot, "packs", id, version);
+            }
+
+            return new PackInfo(new WorkloadPackId(id), version, kind, path, resolvedPackageId);
+        }
+
+        FileInfo PackRecord(PackInfo pack, string sdkFeatureBand)
```

Comment:
Should we parameterize the v1 and default it to `"v1"` in case we ever need v2?

Reply:
I don't think so.  The `v1` in the path is a future-proofing mechanism that we may never need.  If we do need it, then we would probably be changing more about how the records are stored, so the test code would have to update anyway with more than just a change from `v1` to `v2`.